src/test/java/org/entur/lamassu/model/id/IdBuilderTest.java

```diff
@@ -0,0 +1,228 @@
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import org.junit.jupiter.api.Test;
+
+class IdBuilderTest {
+
+  @Test
+  void testCodespace() {
+    String build = IdBuilder
+      .newInstance()
+      .withCodespace("AAA")
+      .withType("Network")
+      .withValue("123")
+      .build();
+    assertEquals("AAA:Network:123", build);
+  }
+
+  @Test
+  void testInvalidCodespaceInput() {
+    assertThrows(
+      IllegalStateException.class,
+      () -> {
+        IdBuilder.newInstance().withType("Network").withValue("123").build();
+      }
+    );
+    assertThrows(
+      IllegalStateException.class,
+      () -> {
+        IdBuilder.newInstance().withCodespace("AAA").withValue("123").build();
+      }
+    );
+    assertThrows(
+      IllegalStateException.class,
+      () -> {
+        IdBuilder.newInstance().withCodespace("AAA").withType("Network").build();
+      }
+    );
+    assertThrows(
+      IllegalStateException.class,
+      () -> {
+        IdBuilder
+          .newInstance()
+          .withCodespace("AA")
+          .withType("Network")
+          .withValue("123")
+          .build();
+      }
+    );
+    assertThrows(
+      IllegalStateException.class,
+      () -> {
+        IdBuilder
+          .newInstance()
+          .withCodespace("AAAA")
+          .withType("Network")
+          .withValue("123")
+          .build();
+      }
+    );
+    assertThrows(
+      IllegalStateException.class,
+      () -> {
+        IdBuilder
+          .newInstance()
+          .withCodespace("AAA")
+          .withType("")
+          .withValue("123")
+          .build();
+      }
+    );
+    assertThrows(
+      IllegalStateException.class,
+      () -> {
+        IdBuilder
+          .newInstance()
+          .withCodespace("AAA")
+          .withType("Network")
+          .withValue("")
+          .build();
+      }
+    );
+    assertThrows(
+      IllegalStateException.class,
+      () -> {
+        IdBuilder.newInstance().withCodespace("AAA").withType("Network!").build();
+      }
+    );
+  }
+
+  @Test
+  void testValidValueInput() {
+    assertDoesNotThrow(() -> {
+      IdBuilder
+        .newInstance()
+        .withCodespace("AAA")
+        .withType("Vehicle")
+        .withValue("ValidString123@")
+        .build();
+    });
+    assertDoesNotThrow(() -> {
+      IdBuilder
+        .newInstance()
+        .withCodespace("AAA")
+        .withType("Vehicle")
+        .withValue("Another:Valid/String_-")
+        .build();
+    });
+    assertDoesNotThrow(() -> {
+      IdBuilder
+        .newInstance()
+        .withCodespace("AAA")
+        .withType("Vehicle")
+        .withValue("EdgeCase")
+        .build();
+    });
+    assertDoesNotThrow(() -> {
+      IdBuilder
+        .newInstance()
+        .withCodespace("AAA")
+        .withType("Vehicle")
+        .withValue("Another.Valid@String:/_-")
```

Comment:
This value works, as value is validated internally via `validateValue`, and not via `DefaultIdValidator.validate`, I think.

Reply:
Yes, I'm not sure why there was originally two different validation methods ... 🤔

Reply:
This definitely uses the DefaultIdValidator.validateValue